### PR TITLE
Added CMake Build as library

### PR DESCRIPTION
**Pull request by jschmold**
Wanted to use Box2D for a personal experiment of mine, and found myself quite unfamiliar with the overall processes involved with building, and referencing it in the project.

### Things it does 

Builds Box2D as a library (`libBox2D.a`)

Installs lib file to the system's lib dir
 - `-lBox2d` to include the library

Installs headers to the system's include dir
-  the headers are automatically found

### Installed Files
```
/usr/local/lib/libbox2d.a
/usr/local/include/Box2D
/usr/local/include/Box2D/Box2D.h
/usr/local/include/Box2D/Collision
/usr/local/include/Box2D/Collision/b2TimeOfImpact.h
/usr/local/include/Box2D/Collision/b2Collision.h
/usr/local/include/Box2D/Collision/b2DynamicTree.h
/usr/local/include/Box2D/Collision/b2Distance.h
/usr/local/include/Box2D/Collision/b2BroadPhase.h
/usr/local/include/Box2D/Collision/Shapes
/usr/local/include/Box2D/Collision/Shapes/b2Shape.h
/usr/local/include/Box2D/Collision/Shapes/b2CircleShape.h
/usr/local/include/Box2D/Collision/Shapes/b2PolygonShape.h
/usr/local/include/Box2D/Collision/Shapes/b2EdgeShape.h
/usr/local/include/Box2D/Collision/Shapes/b2ChainShape.h
/usr/local/include/Box2D/Dynamics
/usr/local/include/Box2D/Dynamics/b2Island.h
/usr/local/include/Box2D/Dynamics/Contacts
/usr/local/include/Box2D/Dynamics/Contacts/b2EdgeAndCircleContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2CircleContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2ContactSolver.h
/usr/local/include/Box2D/Dynamics/Contacts/b2Contact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2PolygonContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2PolygonAndCircleContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2EdgeAndPolygonContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2ChainAndPolygonContact.h
/usr/local/include/Box2D/Dynamics/Contacts/b2ChainAndCircleContact.h
/usr/local/include/Box2D/Dynamics/b2TimeStep.h
/usr/local/include/Box2D/Dynamics/b2Body.h
/usr/local/include/Box2D/Dynamics/b2Fixture.h
/usr/local/include/Box2D/Dynamics/b2WorldCallbacks.h
/usr/local/include/Box2D/Dynamics/b2ContactManager.h
/usr/local/include/Box2D/Dynamics/b2World.h
/usr/local/include/Box2D/Dynamics/Joints
/usr/local/include/Box2D/Dynamics/Joints/b2MotorJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2FrictionJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2WheelJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2PrismaticJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2DistanceJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2Joint.h
/usr/local/include/Box2D/Dynamics/Joints/b2RevoluteJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2GearJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2RopeJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2PulleyJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2MouseJoint.h
/usr/local/include/Box2D/Dynamics/Joints/b2WeldJoint.h
/usr/local/include/Box2D/Common
/usr/local/include/Box2D/Common/b2BlockAllocator.h
/usr/local/include/Box2D/Common/b2Settings.h
/usr/local/include/Box2D/Common/b2GrowableStack.h
/usr/local/include/Box2D/Common/b2Math.h
/usr/local/include/Box2D/Common/b2StackAllocator.h
/usr/local/include/Box2D/Common/b2Timer.h
/usr/local/include/Box2D/Common/b2Draw.h
/usr/local/include/Box2D/Rope
/usr/local/include/Box2D/Rope/b2Rope.h
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,108 @@
+cmake_minimum_required(VERSION 3.12)
```
Comment (jhasse) [inline CMakeLists.txt]: This can be lowered to 3.6 at least, couldn't it?

Comment (jschmold) [reply CMakeLists.txt]: Certainly :smile:. I just tested it with 3.0, and seems to be fine.

**Comment by jschmold** [comment]
Another thing to note, this works splendidly with git submodules as a dependency manager

**Comment by jhasse** [comment]
Yes! Using it in action here: https://gitlab.com/jhasse/ggj2019/tree/master/subprojects

**Comment by erincatto** [comment]
Thanks, I plan to convert to cmake soonish since it has become more standard than premake. I did not have luck with git submodules. GitHub does not include the submodule code in releases.

**Comment by erincatto** [comment]
I will likely setup up Box2D cmake and dependencies in a similar fashion to https://github.com/erincatto/box2d-lite

**Comment by jschmold** [comment]
> Thanks, I plan to convert to cmake soonish since it has become more standard than premake. I did not have luck with git submodules. GitHub does not include the submodule code in releases.

I've done a decent amount of fiddling with CMake. I'd recommend doing like SFML did and having modules, where CMake can look at individual components.


```
find_package(SFML 2.5 COMPONENTS window graphics REQUIRED)
target_link_libraries(project.run 
    sfml-system
    sfml-window
    sfml-graphics
)
```

**Comment by jschmold** [comment]
Sorry for the necro-post, but I believe this is worth mentioning 

https://github.com/felaugmar/Box2D/commit/7f84e6b286eb2755220eedbdf178eb418384ecfa

This fella appears to know _vastly_ more about CMake than I do.

**Comment by erincatto** [comment]
Box2D now supports cmake.

**Pull request closed without merging**